test/integration/scheduler/queue_test.go

```diff
@@ -376,6 +376,44 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			},
 			wantRequeuedPods: sets.New("pod2"),
 		},
+		{
+			name:         "Pod rejected with hostport by the NodePorts plugin is requeued when pod with common hostport is deleted",
```

Comment:
What about adding a pod that won't be requeued even after the pod1 deletion?

Reply:
@macsko 
Thank you for reviewing.
I would like to create a test case like the following:

```go
{
	name:         "Pod rejected with hostport by the NodePorts plugin is requeued when pod with common hostport is deleted",
	initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
	initialPods: []*v1.Pod{
		st.MakePod().Name("pod1").Container("image").ContainerPort([]v1.ContainerPort{{ContainerPort: 8080, HostPort: 8080}}).Node("fake-node").Obj(),
		st.MakePod().Name("pod2").Container("image").ContainerPort([]v1.ContainerPort{{ContainerPort: 8080, HostPort: 8081}}).Node("fake-node").Obj(),
	},
	pods: []*v1.Pod{
		st.MakePod().Name("pod3").Container("image").ContainerPort([]v1.ContainerPort{{ContainerPort: 8080, HostPort: 8080}}).Obj(),
		st.MakePod().Name("pod4").Container("image").ContainerPort([]v1.ContainerPort{{ContainerPort: 8080, HostPort: 8081}}).Obj(),
	},
	triggerFn: func(testCtx *testutils.TestContext) error {
		if err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Delete(testCtx.Ctx, "pod1", metav1.DeleteOptions{GracePeriodSeconds: new(int64)}); err != nil {
			return fmt.Errorf("failed to delete Pod: %w", err)
		}
		return nil
	},
	wantRequeuedPods:          sets.New("pod3"),
	enableSchedulingQueueHint: []bool{true},
}
```

With the above code, we can verify two things and create a test that expects `wantRequeuedPods` to be `sets.New("pod3")`:
- `pod3` has common port, returning `Queue`
- `pod4` does not have common port, returning `QueueSkip`

However, currently, it's not possible to specify multiple `initialPods` (since actually `initialPod *v1.Pod`). As a result, I cannot write the code as shown above. Furthermore, if I define pods with different ports in the `pods` variable, those pods will become schedulable, which does not meet the preconditions for the test (`pods` are expected to be unschedulable at first).

Could you provide any advice on this?

Reply:
@saku3 Feel free to just change `initialPod` to have multiple pods `initialPods`.

Reply:
Thank you for reviewing.
Updated, could you check this, please?